### PR TITLE
feat: add mutation to delete healthcare professional

### LINKED ISSUE #845: Add delete mutation for healthcare professional to the `healthcareProfessionalStore.ts`
# Problem
We have the mutation on the backend set up to delete healthcare professions. We do not have a function to do this set up on the frontend. This will be important going forward due to the possibilities of doctors retiring or going on a sabbatical where they are not actively practicing at an facility.

# Success
The delete functionality for healthcare professionals is added to the `healthcareProfessionalStore.ts`
### Requirements

- [ ] Add the functionality to the `healthcareProfessionalStore.ts`
- [ ] Make sure is accepting the id of the healthcare professional as a parameter.

### PR DESCRIPTION
**Pull request by NabbeunNabi**
Resolves #845 
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specs

## 🔧 What changed
A mutation was added to delete a healthcare professional. This is important as we will in the future need to delete healthcare professionals that retire or go on sabbatical and are not actively practicing.
## 🧪 Testing instructions
Tests will be added after the implementation of ticket #832

### CODE REVIEW

File: stores/healthcareProfessionalsStore.ts [hunk 1]
```diff
@@ -92,3 +107,12 @@ mutation Mutation($updateHealthcareProfessionalId: ID!, $input: UpdateHealthcare
   }
 }
 `
+
+const deleteHealthcareProfessionalGqlMutation = gql`
+mutation Mutation($deleteHealthcareProfessionalId: ID!) {
+  deleteHealthcareProfessional(id: $deleteHealthcareProfessionalId) {
+    isSuccessful
```
Comment (ermish) [inline stores/healthcareProfessionalsStore.ts:114]: We should use this response

Comment (ermish) [reply stores/healthcareProfessionalsStore.ts:114]: It should he used in the caller of the delete action. We should include it in the return type

**Review by ermish — CHANGES_REQUESTED** [review]
Naisu

File: stores/healthcareProfessionalsStore.ts [hunk 2]
```diff
@@ -27,10 +29,23 @@ export const useHealthcareProfessionalsStore = defineStore(
             }
         }
 
+        async function deleteHealthcareProfessional(healthcareProfessionalId: MutationDeleteHealthcareProfessionalArgs) {
+            try {
+                return await graphQLClientRequestWithRetry(
```
Comment (NabbeunNabi) [inline stores/healthcareProfessionalsStore.ts:34]: @ermish I am returning it here. Do you mean specify the type?

File: stores/healthcareProfessionalsStore.ts (hunk 1 above)
Comment (NabbeunNabi) [reply stores/healthcareProfessionalsStore.ts:114]: @ermish I commented where I am using it on the line. For clarification 🙇

**Review by ermish — APPROVED** [review]
ship it!

**Pull request merged**